corehq/apps/geospatial/views.py

```diff
@@ -473,9 +475,20 @@ def post(self, request, domain, *args, **kwargs):
             for child_case_id, parent_case_id in child_to_parent_case_id.items():
                 self._add_related_case(case_id_to_owner_id, child_case_id, parent_case_id)
 
-        update_cases_owner(domain, case_id_to_owner_id)
+        if len(case_id_to_owner_id) <= self.SYNC_CASES_UPDATE_THRESHOLD:
+            update_cases_owner(domain, case_id_to_owner_id)
+            return JsonResponse({'success': True, 'message': _('Cases were reassigned successfully')})
+        else:
+            return self.process_as_async(case_id_to_owner_id)
 
-        return JsonResponse({'success': True, 'message': _('Cases were reassigned successfully')})
+    def process_as_async(self, case_id_to_owner_id):
+        geo_cases_reassignment_update_owners.apply_async((self.domain, case_id_to_owner_id))
+        return JsonResponse(
+            {
+                'success': True,
+                'message': _('Cases reassignment request is accepted and will be completed in some time')
```

Comment:
Nit: Small grammar change
```suggestion
                'message': _('Case reassignment request has been accepted and will be completed in some time')
```

Reply:
Addressed in https://github.com/dimagi/commcare-hq/commit/454a410869284fc05ecfee94bdb0d8e1c3cf2709